#! /bin/sh /usr/share/dpatch/dpatch-run
## Paul Brossier <piem@debian.org>
## DP: fix segfault when dvdauth chapter list empty

@DPATCH@
--- kino-0.76.orig/src/page_export_mjpeg.cc
+++ kino-0.76/src/page_export_mjpeg.cc
@@ -472,8 +472,10 @@
 		{
 			std::string tmpSceneStr;
 			tmpSceneStr = this->common->getTime().parseFramesToString( *sceneIter, SMIL::Time::TIME_FORMAT_SMPTE );
+                        if (tmpSceneStr.length() > 9)
 			chapterList << "," << tmpSceneStr.substr( 1, 7 ) << "." << std::setw( 2 ) << std::setfill( '0' ) <<
 				static_cast< int >( atof( tmpSceneStr.substr( 9, 2 ).c_str() ) / ( isPal ? 25 : 29.97 ) * 100 );
+                        else goto endchapter;
 		}
 
 		/* output the single vob, checking for chapter marks */
@@ -481,6 +483,8 @@
 		chapterList.str() << "\" pause=\"0\" />" << std::endl;
 	}
 
+endchapter:
+
 	xml << "\t\t\t</pgc>" << std::endl;
 	xml << "\t\t</titles>" << std::endl;
 	xml << "\t</titleset>" <<std::endl;
